vault12factor/__init__.py

```diff
@@ -226,10 +226,17 @@ def __init__(self, vaulturl: str, vaultauth: VaultAuthentication, secretpath: st
         self.pin_cacert = pin_cacert
         self.ssl_verify = ssl_verify
         self.debug_output = debug_output
-        self._cache = None  # type: Dict[str, str]
+        self._cache = {}  # type: Dict[str, str]
         self._leasetime = None  # type: datetime.datetime
         self._updatetime = None  # type: datetime.datetime
         self._lease_id = None  # type: str
+        self._refresh()
+
+    def _attach_secret_attribute(instance, attribute):
```

Comment:
* Please add PEP484 type annotations for `attribute` and the return value (`None`)
* Please don't override the `self` convention (as [per PEP-8](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments))